dipy/reconst/ivim.py

```diff
@@ -0,0 +1,541 @@
+
+    Returns
+    -------
+    S : array
+        An array containing the IVIM signal estimated using given parameters.
+    """
+    S0, f, D_star, D = params
+    b = gtab.bvals
+    S = S0 * (f * np.exp(-b * D_star) + (1 - f) * np.exp(-b * D))
+    return S
+
+
+def _ivim_error(params, gtab, signal):
+    """Error function to be used in fitting the IVIM model.
+
+    Parameters
+    ----------
+    params : array
+        An array of IVIM parameters - [S0, f, D_star, D]
+
+    gtab : GradientTable class instance
+        Gradient directions and bvalues.
+
+    signal : array
+        Array containing the actual signal values.
+
+    Returns
+    -------
+    residual : array
+        An array containing the difference between actual and estimated signal.
+    """
+    residual = signal - ivim_prediction(params, gtab)
+    return residual
+
+
+def f_D_star_prediction(params, gtab, S0, D):
+    """Function used to predict IVIM signal when S0 and D are known
+    by considering f and D_star as the unknown parameters.
+
+    Parameters
+    ----------
+    params : array
+        The value of f and D_star.
+
+    gtab : GradientTable class instance
+        Gradient directions and bvalues.
+
+    S0 : float
+        The parameters S0 obtained from a linear fit.
+
+    D : float
+        The parameters D obtained from a linear fit.
+
+    Returns
+    -------
+    S : array
+        An array containing the IVIM signal estimated using given parameters.
+    """
+    f, D_star = params
+    b = gtab.bvals
+    S = S0 * (f * np.exp(-b * D_star) + (1 - f) * np.exp(-b * D))
+    return S
+
+
+def f_D_star_error(params, gtab, signal, S0, D):
+    """Error function used to fit f and D_star keeping S0 and D fixed
+
+    Parameters
+    ----------
+    params : array
+        The value of f and D_star.
+
+    gtab : GradientTable class instance
+        Gradient directions and bvalues.
+
+    signal : array
+        Array containing the actual signal values.
+
+    S0 : float
+        The parameters S0 obtained from a linear fit.
+
+    D : float
+        The parameters D obtained from a linear fit.
+
+    Returns
+    -------
+    residual : array
+        An array containing the difference of actual and estimated signal.
+    """
+    f, D_star = params
+    return signal - f_D_star_prediction([f, D_star], gtab, S0, D)
+
+
+class IvimModel(ReconstModel):
+    """Ivim model
+    """
+
+    def __init__(self, gtab, split_b_D=400.0, split_b_S0=200., bounds=None,
+                 tol=1e-15, f_threshold=0.25, x_scale=[1000., 0.1, 0.001, 0.0001],
+                 options={'gtol': 1e-15, 'ftol': 1e-15,
+                          'eps': 1e-15, 'maxiter': 1000}):
+        """
+        Initialize an IVIM model.
+
+        The IVIM model assumes that biological tissue includes a volume
+        fraction 'f' of water flowing in perfused capillaries, with a
+        perfusion coefficient D* and a fraction (1-f) of static (diffusion
```

Comment:
I'd change this to "... of water flowing, with a pseudo-perfusion coefficient D*..."

It's not truly the perfusion.